tests/unit/amo/components/TestAddonRecommendations.js

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+
+import AddonRecommendations, {
+  AddonRecommendationsBase,
+  TAAR_IMPRESSION_CATEGORY,
+  TAAR_COHORT_COOKIE_NAME,
+  TAAR_COHORT_INCLUDED,
+  TAAR_COHORT_EXCLUDED,
+} from 'amo/components/AddonRecommendations';
+import { createInternalAddon } from 'core/reducers/addons';
+import {
+  fakeAddon,
+} from 'tests/unit/amo/helpers';
+import { fakeI18n, shallowUntilTarget } from 'tests/unit/helpers';
+
+
+const fakeCookie = (returnValue) => {
+  return {
+    load: sinon.stub().returns(returnValue),
+    save: sinon.spy(),
+  };
+};
+
+const fakeRandomizer = (returnValue) => {
+  return sinon.stub().returns(returnValue);
+};
+
+describe(__filename, () => {
+  let fakeTracking;
+
+  beforeEach(() => {
+    fakeTracking = {
+      sendEvent: sinon.spy(),
+    };
+  });
+
+  function render({
+    addon = createInternalAddon(fakeAddon),
+    _cookie = fakeCookie(),
+    _randomizer = fakeRandomizer(0),
+    ...props
+  } = {}) {
+    return shallowUntilTarget(
+      <AddonRecommendations
+        addon={addon}
+        i18n={fakeI18n()}
+        _cookie={_cookie}
+        _randomizer={_randomizer}
+        _tracking={fakeTracking}
+        {...props}
+      />,
+      AddonRecommendationsBase
+    );
+  }
+
+  it('renders nothing without an addon', () => {
+    const root = render({ addon: null });
+    expect(root).not.toHaveClassName('AddonRecommendations');
+  });
+
+  it('uses an existing cookie to determine the variant', () => {
+    const cohort = TAAR_COHORT_INCLUDED;
+    const _cookie = fakeCookie(cohort);
+    const root = render({ _cookie });
+    expect(root).toHaveClassName('AddonRecommendations');
+    expect(root.render().find('h1').hasClass(cohort)).toEqual(true);
+  });
+
+  it('uses the randomizer to set the cohort and cookie when a cookie does not exist', () => {
+    const checkWithRandomizer = (randomNumber, expectedCohort) => {
```

Comment:
Maybe `checkWithFixedRandomizer`?

Reply:
(could not come up with something better, yet I find this function name a bit obscure)